### PR TITLE
tcl-9_0,tk-9_0: init

### PR DESCRIPTION
**Pull request by fgaz**
## Description of changes

https://www.tcl-lang.org/software/tcltk/9.0.html

8.6 is still maintained, let's wait a bit before changing the default

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fgaz** [comment]
@agbrooks ping

File: pkgs/development/interpreters/tcl/generic.nix [hunk 1]
```diff
@@ -24,17 +24,39 @@ let
...
         "--enable-threads"
+      ] ++ [
         # Note: using $out instead of $man to prevent a runtime dependency on $man.
         "--mandir=${placeholder "out"}/share/man"
+      ] ++ lib.optionals (lib.versionOlder version "9.0") [
         "--enable-man-symlinks"
         # Don't install tzdata because NixOS already has a more up-to-date copy.
         "--with-tzdata=no"
+      ] ++ lib.optionals (lib.versionAtLeast version "9.0") [
+        # By default, tcl libraries get zipped and embedded into libtcl*.so,
```
Comment (agbrooks) [inline pkgs/development/interpreters/tcl/generic.nix:51]: This is a very helpful comment. Thank you for adding it!

**Review by agbrooks — APPROVED** [review]
My apologies for the late review. This looks reasonable; thanks for adding it!

**Comment by agbrooks** [comment]
> @agbrooks are you still maintaining the tcl package? looks like 8.6 is stuck on a point release from 2022

Yes, just had my hands pretty full recently. Thanks for the reminder! I'd be happy to approve a PR updating it, otherwise, I suspect I'll have a chance to address that this weekend.

**Comment by fgaz** [comment]
Great! And no rush. It worked well for two years, surely it can wait a few weeks.

By the way, could you also take a look at #344716? I'd like to merge it before the 24.11 branch-off

**Pull request merged**